pkg/kubelet/kubelet_node_status.go

```diff
@@ -761,7 +761,9 @@ func (kl *Kubelet) defaultNodeStatusFuncs() []func(context.Context, *v1.Node) er
 	// if cloud is not nil, we expect the cloud resource sync manager to exist
 	var nodeAddressesFunc func() ([]v1.NodeAddress, error)
 	if kl.cloud != nil {
-		nodeAddressesFunc = kl.cloudResourceSyncManager.NodeAddresses
+		nodeAddressesFunc = func() ([]v1.NodeAddress, error) {
+			return kl.cloudResourceSyncManager.NodeAddresses(context.TODO())
+		}
```

Comment:
Did you try to change `NodeAddress` to accept a function with context as a first parameter?

Reply:
Is this change required? If so, why?

Reply:
This is not resolved yet, please unmark.